---
 Core/Src/stm32h7xx_it.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Core/Src/stm32h7xx_it.c b/Core/Src/stm32h7xx_it.c
index c63be4c..0ff7307 100644
--- a/Core/Src/stm32h7xx_it.c
+++ b/Core/Src/stm32h7xx_it.c
@@ -51,8 +51,8 @@ uint32_t Sweep_step_count = 0;
 float Sweep_time_elapsed = 0.0f;
 
 uint8_t IDLE_flag = 0;
-uint32_t angleInner;
-uint32_t angleOutter;
+volatile uint32_t angleInner;
+volatile uint32_t angleOutter;
 float angleInnerFloat;
 float angleOutterFloat;
 
-- 
2.45.2.windows.1

